src/main/java/core/util/HODateTime.java

```diff
@@ -23,50 +25,74 @@
     /**
      * the birthday of hattrick
      */
-    private static final HODateTime htStart = HODateTime.fromHT("1997-09-22 00:00:00");
+    public static final HODateTime htStart = HODateTime.fromHT("1997-09-22 00:00:00");
 
     /**
      * internal time representation
      */
-    public Instant instant;
+    final public Instant instant;
 
     /**
      * create an HODateTime instance (should it be private?)
+     *
      * @param in Instant
      */
-    public HODateTime(Instant in) {
+    public HODateTime(@NotNull Instant in) {
         this.instant = in;
     }
+    public HODateTime(@NotNull HODateTime in){this.instant=in.instant;}
 
     /**
      * Create instance from HT (chpp) string
+     *
      * @param htString HT string
      * @return HODateTime
      */
     public static HODateTime fromHT(String htString) {
-        LocalDateTime htTime = LocalDateTime.parse(htString, cl_Formatter);
-        return new HODateTime(htTime.atZone(DEFAULT_TIMEZONE).toInstant());
+        if ( htString != null && !htString.isEmpty()) {
+            LocalDateTime htTime = LocalDateTime.parse(htString, cl_Formatter);
+            return new HODateTime(htTime.atZone(DEFAULT_TIMEZONE).toInstant());
+        }
+        return null;
     }
 
     /**
      * Create an instance from database timestamp
+     *
      * @param timestamp database timestamp
      * @return HODateTime
      */
     public static HODateTime fromDbTimestamp(Timestamp timestamp) {
-        return new HODateTime(timestamp.toInstant());
+        if ( timestamp != null )return new HODateTime(timestamp.toInstant());
```

Comment:
Bit of a funky formatting here.  Depending on your keymap in IntelliJ, the shortcut to Reformat code will vary – for me it's Alt+Shift+F.

Reply:
changed